src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -139,32 +138,30 @@ internal Metadata GetMetadata(string fullPath, DateTime snapshotTimestamp)
         if (_metadataCache.TryGetMetadata(key, out var metadata))
             return metadata;
 
-        var newMetadata = GetMetadataWorker();
+        // here, we don't care about timestamp since all those bits should be part of Fx. and we assume that 
+        // it won't be changed in the middle of VS running.
+        var (newMetadata, handles) = GetMetadataWorker(fullPath);
 
         if (!_metadataCache.GetOrAddMetadata(key, newMetadata, out metadata))
             newMetadata.Dispose();
 
-        return metadata;
+        if (handles != null)
+            s_metadataToStorageHandles.Add(newMetadata, handles);
```

Comment:
pulled the logic up to here for populating the caches.  it simplified the lower code (imo).